domain/machine/service/watcher.go

```diff
@@ -81,14 +81,28 @@ func (s *WatchableService) WatchModelMachines() (watcher.StringsWatcher, error)
 	)
 }
 
-// WatchMachineCloudInstances returns a StringsWatcher that is subscribed to
-// the changes in the machine_cloud_instance table in the model.
-func (s *WatchableService) WatchMachineCloudInstances(ctx context.Context) (watcher.StringsWatcher, error) {
-	table, stmt := s.st.InitialWatchInstanceStatement()
-	return s.watcherFactory.NewNamespaceWatcher(
-		table,
+// WatchMachineCloudInstances returns a NotifyWatcher that is subscribed to
+// the changes in the machine_cloud_instance table in the model, for the given
+// machine UUID.
+func (s *WatchableService) WatchMachineCloudInstances(ctx context.Context, machineUUID string) (watcher.NotifyWatcher, error) {
+	return s.watcherFactory.NewNamespaceNotifyMapperWatcher(
+		"machine_cloud_instance",
 		changestream.All,
-		eventsource.InitialNamespaceChanges(stmt),
+		eventsource.FilterEvents(func(event changestream.ChangeEvent) bool {
+			return event.Changed() == machineUUID
+		}),
+	)
+}
+
+// WatchLXDProfiles returns a NotifyWatcher that is subscribed to the changes in
+// the machine_cloud_instance table in the model, for the given machine UUID.
+func (s *WatchableService) WatchLXDProfiles(ctx context.Context, machineUUID string) (watcher.NotifyWatcher, error) {
+	return s.watcherFactory.NewNamespaceNotifyMapperWatcher(
+		"machine_lxd_profile",
+		changestream.All,
+		eventsource.FilterEvents(func(event changestream.ChangeEvent) bool {
+			return event.Changed() == machineUUID
+		}),
```

Comment:
Can you add a note that this is currently driven off the `machine_lxd_profiles`, but we might want to drive this using a logical change stream event instead?

The reason is, that if we have a lot of charms adding a lot of profiles, then this is going to be a noisy table. We can instead drive the watcher on a logical change stream event when we call `SetAppliedLXDProfileNames`.

Reply:
Sure, done!